### PR TITLE
Support for Jasmine sessions for BrowserStack Test Observability

### PR DESCRIPTION
**Pull request by nagpalkaran95**
## Proposed changes

[//]: # Add support for Jasmine sessions to be captured by BrowserStack Test Observability

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

File: packages/wdio-browserstack-service/src/util.ts [hunk 1]
```diff
@@ -550,4 +553,17 @@ export function getObservabilityBuildTags(options: BrowserstackConfig & Options.
...
+    if (framework === 'mocha') {
+        if (hook === 'before' || hook === 'after' || hook === 'beforeEach' || hook === 'afterEach') {
+            return true
+        }
+    } else if (framework === 'jasmine') {
+        if (hook === 'beforeEach' || hook === 'afterEach') {
+            return true
+        }
+    }
+    return false
```
Comment (christian-bromann) [inline packages/wdio-browserstack-service/src/util.ts:566]:
```suggestion
    if (framework === 'mocha' && hook === 'before' || hook === 'after' || hook === 'beforeEach' || hook === 'afterEach') {
        return true
    }

    if (framework === 'jasmine' && hook === 'beforeEach' || hook === 'afterEach') {
        return true
    }

    return false
```

**Review by christian-bromann — APPROVED** [review]
LGTM 👍

**Comment by christian-bromann** [comment]
@nagpalkaran95 it seems unit tests are failing, mind taking a look?

**Comment by sriteja777** [comment]
@christian-bromann Have fixed the failing tests.

**Review by christian-bromann — APPROVED** [review]
LGTM 👍  thank you all!

**Pull request merged**